test/mocks/index.ts

```diff
@@ -0,0 +1,218 @@
+import nock from "nock";
+import { expect } from "chai";
+import { numberPurchasableByAccount } from "../../src/actions/sale/numberPurchasableByAccount";
+import { Mintlist, SaleContractConfig } from "../../src/types";
+import { Sale } from "../../src/contracts/types";
+import { SalePhase } from "../../src/types";
+import "mocha";
+import {
+  BigNumber,
+  CallOverrides,
+  ContractTransaction,
+  ethers,
+  Signer,
+} from "ethers";
+import { getMintlist } from "../../src/actions/sale/getMintlist";
+import {
+  getSaleData,
+  getSaleStatus,
+  purchaseDomains,
+} from "../../src/actions/sale";
+import { Block, BlockTag, Provider } from "@ethersproject/providers";
+import { SalePhases } from "../../src/constants";
+
+export const feb23TsSeconds = 1677173138;
+export const feb24TsSeconds = 1677264314;
+
+export type Mocks = {
+  signer: Partial<Signer>;
+  sale: Partial<Sale>;
+  currentSaleConfiguration: [
+    string,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    BigNumber,
+    string,
+    BigNumber,
+    BigNumber,
+    BigNumber
+  ] & {
+    sellerWallet: string;
+    parentDomainId: BigNumber;
+    salePrice: BigNumber;
+    privateSalePrice: BigNumber;
+    mintlistSaleDuration: BigNumber;
+    amountForSale: BigNumber;
+    mintlistMerkleRoot: string;
+    startingMetadataIndex: BigNumber;
+    folderGroupID: BigNumber;
+    publicSaleLimit: BigNumber;
+  };
+  currentSalePhase: number;
+  currentDomainsSold: BigNumber;
+  currentSaleStartTimestamp: BigNumber;
+  currentLatestBlock: Partial<Block>;
+  currentSaleId: string;
+  currentSaleCounter: BigNumber;
+  currentPauseStatus: boolean;
+  currentSignerFunds: ethers.BigNumber;
+  currentDomainsPurchasedByAccountPerSale: BigNumber;
```

Comment:
We know everything applies to `current` sale, so calling everything `current` is redundant. Please remove it